dev/puppet-server.conf.sample

```diff
@@ -30,8 +30,11 @@ webserver: {
 # configure the mount points for the web apps
 web-router-service: {
     "puppetlabs.services.ca.certificate-authority-service/certificate-authority-service": "/puppet-ca"
-    "puppetlabs.services.master.master-service/master-service": "/puppet"
+    "puppetlabs.services.master.master-service/master-service": {master-routes:       "/puppet"
 
+                                                                 # The following controls 404 errors and should not
+                                                                 # be modified
+                                                                 invalid-in-puppet-4: "/" }
```

Comment:
Oh wait.  What does the user see if they request `/some/resource/that/never/existed/ever` (And a previous Puppet agent never tried to request)?

If they see the "upgrade" error message, then that's an issue we should discuss further, because that request isn't being made by an agent that needs to be upgraded.